File: lz-commons/src/main/java/com/eucalypt/lz/commons/beans/SystemSettingService.java
package com.eucalypt.lz.commons.beans;

import com.eucalypt.framework.core.service.AbstractService;
import org.springframework.stereotype.Component;

/**
 * Created by zhaoyang on 17/7/2.
 */
@Component
public class SystemSettingService extends AbstractService<SystemSetting> {

	@Override
	protected Class<SystemSetting> getEntityClass() {
		return SystemSetting.class;
	}

	public String save(SystemSetting setting) {
		return super.save(setting);
	}

	public SystemSetting get() {
		SystemSetting systemSetting = super.get();
		if (systemSetting == null) {
			synchronized (this) {
				systemSetting = new SystemSetting();
				super.save(systemSetting);
			}
		}
		return systemSetting;
	}
}

File: lz-admin/src/main/webapp/js/modules/sys/services/SystemSettingService.js
define(function () {
	return ['RestService', SystemSettingService];
	function SystemSettingService(RestService) {
		return {
			save: function (setting) {
				return RestService.post('sys/setting/save', setting);
			},
			get: function () {
				return RestService.post('sys/setting/get');
			}
		};
	}
});